src/utilities/effects.ts

```diff
@@ -3840,6 +3888,24 @@ const effects = {
     }
   },
 
+  async getFile(
+    fileId: number,
+    user: User | null,
+    signal: AbortSignal | undefined = undefined,
+  ): Promise<{ aborted: boolean; file: string | null }> {
+    try {
+      const file = await reqGateway(`/file/${fileId}`, 'GET', null, user, true, signal, false);
+      return { aborted: false, file };
+    } catch (e) {
+      if ((e as Error).name === 'AbortError') {
+        return { aborted: true, file: null };
+      } else {
+        catchError(e as Error);
```

Comment:
Should this also trigger the failure toast?

Reply:
Ah...I think that must have be an oversight... :facepalm: I'd imagine it would be more helpful to have one than to not. If you agree, would you mind adding to both? Otherwise, I'm fine with not adding bloat to this PR.

Reply:
Added!